jsr352/core/src/main/java/org/hibernate/search/jsr352/logging/impl/Log.java

```diff
@@ -0,0 +1,227 @@
+
+	@Message(id = JSR_352_MESSAGES_START_ID + 3,
+			value = "Unknown entity manager factory scope: '%1$s'. Please use a supported scope.")
+	SearchException unknownEntityManagerFactoryScope(String scopeName);
+
+	@Message(id = JSR_352_MESSAGES_START_ID + 4,
+			value = "Exception while retrieving the EntityManagerFactory using @PersistenceUnit."
+					+ " This generally happens either because the persistence wasn't configured properly"
+					+ " or because there are multiple persistence units."
+	)
+	SearchException cannotRetrieveEntityManagerFactoryInJsr352();
+
+	@Message(id = JSR_352_MESSAGES_START_ID + 5,
+			value = "Multiple entity manager factories have been registered in the CDI context."
+					+ " Please provide the bean name for the selected entity manager factory to the batch indexing job through"
+					+ " the 'entityManagerFactoryReference' parameter."
+	)
+	SearchException ambiguousEntityManagerFactoryInJsr352();
+
+	@Message(id = JSR_352_MESSAGES_START_ID + 6,
+			value = "No entity manager factory has been created with this persistence unit name yet: '%1$s'."
+					+ " Make sure you use the JPA API to create your entity manager factory (use a 'persistence.xml' file)"
+					+ " and that the entity manager factory has already been created and wasn't closed before"
+					+ " you launch the job."
+	)
+	SearchException cannotFindEntityManagerFactoryByPUName(String persistentUnitName);
+
+	@Message(id = JSR_352_MESSAGES_START_ID + 7,
+			value = "No entity manager factory has been created with this name yet: '%1$s'."
+					+ " Make sure your entity manager factory is named (for instance by setting the '%2$s' option)"
+					+ " and that the entity manager factory has already been created and wasn't closed before"
+					+ " you launch the job."
+	)
+	SearchException cannotFindEntityManagerFactoryByName(String entityManagerFactoryName, String option);
+
+	@Message(id = JSR_352_MESSAGES_START_ID + 8,
+			value = "No entity manager factory has been created yet."
+					+ " Make sure that the entity manager factory has already been created and wasn't closed before"
+					+ " you launched the job."
+	)
+	SearchException noEntityManagerFactoryCreated();
+
+	@Message(id = JSR_352_MESSAGES_START_ID + 9,
+			value = "Multiple entity manager factories are currently active."
+					+ " Please provide the name of the selected persistence unit to the batch indexing job through"
+					+ " the 'entityManagerFactoryReference' parameter (you may also use the 'entityManagerFactoryScope'"
+					+ " parameter for more referencing options)."
+	)
+	SearchException tooManyActiveEntityManagerFactories();
+
+	@LogMessage(level = INFO)
+	@Message(id = JSR_352_MESSAGES_START_ID + 10,
+			value = "%1$s"
+	)
+	void analyzeIndexProgress(StringBuilder stringBuilder);
+
+	@LogMessage(level = DEBUG)
+	@Message(id = JSR_352_MESSAGES_START_ID + 11,
+			value = "Session is closed."
+	)
+	void sessionClosed();
```

Comment:
I don't see the point of those "xxxClosed" methods. We expect the job to close resources correctly, so we should only log stuff when the job *cannot* close it. I think we should those "xxxClosed" methods; same for `closingEverything`.